Authorino crashes when external OPA response misses `Content-Type`

When using OPA authorization and `externalRegistry` to source the Rego policy, if the external policy registry responds with an HTTP response that misses the `Content-Type` header, Authorino fails to parse the response and the pod crashes severely.

Logs from the Authorino pod:

```
{"level":"info","ts":1661328961.5958905,"logger":"authorino","msg":"setting instance base logger","min level":"info","mode":"production"}
{"level":"info","ts":1661328962.647555,"logger":"authorino","msg":"Waited for 1.02878305s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/cloudcredential.openshift.io/v1?timeout=32s\n"}
{"level":"info","ts":1661328964.4013333,"logger":"authorino.controller-runtime.metrics","msg":"Metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":1661328964.4023006,"logger":"authorino","msg":"starting http auth service","port":"5001","tls":false}
{"level":"info","ts":1661328964.402353,"logger":"authorino","msg":"starting http oidc service","port":"8083","tls":false}
{"level":"info","ts":1661328964.4024544,"logger":"authorino","msg":"starting grpc auth service","port":"50051","tls":false}
{"level":"info","ts":1661328964.4024792,"logger":"authorino","msg":"starting manager"}
{"level":"info","ts":1661328964.4042392,"logger":"authorino","msg":"Starting server","path":"/metrics","kind":"metrics","addr":"[::]:8080"}
{"level":"info","ts":1661328964.4047065,"logger":"authorino.controller.authconfig","msg":"Starting EventSource","reconciler group":"authorino.kuadrant.io","reconciler kind":"AuthConfig","source":"kind source: *v1beta1.AuthConfig"}
{"level":"info","ts":1661328964.4047604,"logger":"authorino.controller.authconfig","msg":"Starting Controller","reconciler group":"authorino.kuadrant.io","reconciler kind":"AuthConfig"}
{"level":"info","ts":1661328964.4049587,"logger":"authorino.controller.secret","msg":"Starting EventSource","reconciler group":"","reconciler kind":"Secret","source":"kind source: *v1.Secret"}
{"level":"info","ts":1661328964.4050648,"logger":"authorino.controller.secret","msg":"Starting Controller","reconciler group":"","reconciler kind":"Secret"}
{"level":"info","ts":1661328964.5055656,"logger":"authorino.controller.secret","msg":"Starting workers","reconciler group":"","reconciler kind":"Secret","worker count":1}
{"level":"info","ts":1661328964.50553,"logger":"authorino.controller.authconfig","msg":"Starting workers","reconciler group":"authorino.kuadrant.io","reconciler kind":"AuthConfig","worker count":1}
{"level":"info","ts":1661328964.5060177,"logger":"authorino.controller-runtime.manager.controller.authconfig.bootstrap","msg":"building the index","count":1}
panic: runtime error: index out of range [0] with length 0
goroutine 613 [running]:
github.com/kuadrant/authorino/pkg/evaluators/authorization.(*OPAExternalSource).downloadRegoDataFromUrl(0xc0006a4a20)
    /workspace/pkg/evaluators/authorization/opa.go:224 +0xca5
github.com/kuadrant/authorino/pkg/evaluators/authorization.NewOPAAuthorization({0xc0006221e0, 0x3}, {0x0, 0x0}, 0xc0004ec280, 0x0, 0xc000523ef0, {0x1f4f708, 0xc0006a49f0})
    /workspace/pkg/evaluators/authorization/opa.go:44 +0x145
github.com/kuadrant/authorino/controllers.(*AuthConfigReconciler).translateAuthConfig(0xc000338540, {0x1f4f708, 0xc000523e60}, 0xc0005bb200)
    /workspace/controllers/auth_config_controller.go:440 +0x221c
github.com/kuadrant/authorino/controllers.(*AuthConfigReconciler).Reconcile(0xc000338540, {0x1f4f708, 0xc00073c060}, {{{0xc00040a720, 0x1babbe0}, {0xc00040a708, 0x30}}})
    /workspace/controllers/auth_config_controller.go:106 +0x4ce
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc0003b1340, {0x1f4f708, 0xc000361dd0}, {{{0xc00040a720, 0x1babbe0}, {0xc00040a708, 0x413bf4}}})
    /opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114 +0x26f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0003b1340, {0x1f4f660, 0xc00075e680}, {0x1a91780, 0xc000504a80})
    /opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311 +0x33e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0003b1340, {0x1f4f660, 0xc00075e680})
    /opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
    /opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
    /opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:223 +0x357
```

Thanks @jsmolar for spotting it!

--- a/pkg/evaluators/authorization/opa.go
+++ b/pkg/evaluators/authorization/opa.go
@@ -221,7 +221,7 @@ func (ext *OPAExternalSource) downloadRegoDataFromUrl() (string, error) {
 
 		result := string(body)
 		//json
-		if resp.Header["Content-Type"][0] == "application/json" {
+		if strings.Contains(resp.Header.Get("Content-Type"), "application/json") {
 			var jsonResponse responseOpaJson
 			if err := json.Unmarshal(body, &jsonResponse); err != nil {
 				return "", fmt.Errorf("unable to unmarshal json response: %v", err)

--- a/pkg/evaluators/authorization/opa_test.go
+++ b/pkg/evaluators/authorization/opa_test.go
@@ -105,6 +105,25 @@ func TestOPAExternalUrlJsonResponse(t *testing.T) {
 	assertOPAAuthorization(t, opa)
 }
 
+func TestOPAExternalUrlMissingContentType(t *testing.T) {
+	extHttpMetadataServer := httptest.NewHttpServerMock(opaExtHttpServerMockAddr, map[string]httptest.HttpServerMockResponseFunc{
+		"/rego": func() httptest.HttpServerMockResponse {
+			return httptest.HttpServerMockResponse{Status: 200, Body: opaInlineRegoDataMock, Headers: map[string]string{"Content-Type": ""}}
+		},
+	})
+	defer extHttpMetadataServer.Close()
+
+	externalSource := &OPAExternalSource{
+		Endpoint:        "http://" + opaExtHttpServerMockAddr + "/rego",
+		AuthCredentials: auth.NewAuthCredential("", ""),
+	}
+
+	opa, err := NewOPAAuthorization("test-opa", "", externalSource, false, 0, context.TODO())
+
+	assert.NilError(t, err)
+	assertOPAAuthorization(t, opa)
+}
+
 func TestOPAExternalUrlWithTTL(t *testing.T) {
 	changed := false
 	extHttpMetadataServer := httptest.NewHttpServerMock(opaExtHttpServerMockAddr, map[string]httptest.HttpServerMockResponseFunc{

